File: mr-plugin/server/mr-core/mr_core/models/project.py
from __future__ import annotations
from peewee import *
from playhouse.postgres_ext import JSONField
from .base.read_write import ReadWriteModel
from datetime import datetime
import uuid
import pytz


class Project(ReadWriteModel):
    id = UUIDField(primary_key=True, default=uuid.uuid4)  # UUID field with default generation
    name = CharField(null=False, unique=True)
    title = CharField(null=True, unique=True)
    description = TextField(null=True, unique=False)
    created_at = DateTimeField(null=False, default=lambda: datetime.now(pytz.utc))  # Consider using DateTimeField for actual date-time values
    url = CharField()
    data_types = JSONField()  # List of strings
    admin = CharField()
    created_by = CharField()
    readme_md = TextField()  # Markdown content

    class Meta:
        indexes = (
            (('name',), True),
        )

    @classmethod
    def create(cls, user=None, **query):
        return super(Project,cls).create(user, **query)

File: mr-plugin/model-registry-ui/src/api/data_types/project.ts
export interface ProjectInterface {
	id: string;         // UUID
	name: string;       // Name of the project
	title: string;      // Title of the project
	description: string;// Description of the project
	created_at: string; // ISO date string for creation date
	url: string;        // bucket URL
	created_by: string; // Username of the creator
	readme_md: string;     // readme markdown
	roles: object;      // User roles
	
}

export class Project implements ProjectInterface {
	
	id: string;
	name: string;
	title: string;
	description: string;
	created_at: string;
	url: string;
	created_by: string;
	readme_md: string;
	roles: object;
	
	constructor(data: Partial<ProjectInterface>) {
		this.id = data.id || '';
		this.name = data.name || '';
		this.title = data.title || '';
		this.description = data.description || '';
		this.created_at = data.created_at || '';
		this.url = data.url || '';
		this.roles = data.roles || {};
		this.created_by = data.created_by || '';
		this.readme_md = data.readme_md || '';
	}
}
